### DISCUSSION TITLE
Settings that equal video seconds

### DISCUSSION THREAD

**Discussion by gateway** [rkhamilton/vqgan-clip-generator#20] [Q&A] [answered]
I feel really dumb and your probably tired of my questions but I'm trying to figure out the proper formula to create a video thats a length of x.

So in story mode lets say I have 4 stories.. separated by a ^ and lets say I want each story to run 5 seconds long what are the proper settings.. out of the code block here.

https://youtu.be/4YLRfN5G32c

```# Now generate a zoom video starting from that initial frame.
generate.video_frames(eng_config = config,
        text_prompts = text_prompts,
        init_image = init_image+'.png',
        iterations = 4000,
        save_every = 10,
        change_prompt_every = 700)
```

the videos I have done that are longer seem to loop back to the first and second story for some reason.. 

:)

**Comment by rkhamilton** [comment, marked as answer]
Good question, and I hope I can clarify this.

In your example, video_frames will create 4000 iterations. Every 10ths iteration will be saved to disk as a sequentially named PNG. So in this case you end up with 400 frames of video.

The duration of the video is determined when you run encode_video. FFMPEG will take your 400 frames of video and encode them based on the output_framerate parameter. If you wanted to treat that as 24 fps footage, you set output_framerate=24, and then you would get 400/24 = 16.6 seconds of footage.

So the final duration is
**iterations / save_every / output_framerate = seconds of footage**.

You can also add ffmpeg interpolation to that output footage if you use the optional assumed_input_framerate parameter. If you use this, you can speed up or slow down the footage by the ratio of output_framerate / assumed_input_framerate  .
If assumed_input_framerate = 12, and you keep output_framerate = 24, then ffpmeg will interpolate one frame in between each frame of input. So if output_framerate is not equal to assumed_input_framerate, then the duration of the clip is set only by assumed_input_framerate.
**iterations / save_every / assumed_input_framerate = seconds of footage**.

Typing this out, I realize it's not super clear. In v1.2.0 I believe I'll change encode_video to require input_framerate, and make output_framerate optional. With that change the duration would always be **iterations / save_every / input_framerate**, and you would only be able change the video footage interpolation with output_framerate.

**Reply by gateway** [reply]
/me scratches head.. I get it.. I just honestly need to sit down and focus and am juggling to many things at once right now.. but thank you for the answer.  It's a tiny bit confusing.. ;)

**Comment by rkhamilton** [comment]
Sorry, I missed your comment about prompts. Yes, the way story prompts work is that they cycle back to the beginning if you run out of prompts. I don't really use story prompts and don't feel strongly. If you would prefer that they just stick to the last prompt when you get to the end of the list I can make that change.

**Reply by gateway** [reply]
Yea, was just trying a way to figure out if i have x story prompts and want 20 seconds of them it would device 20 by x to do each prompt for a number of seconds.. having them loop back is kinda annoying and I would have to pull it up into adobe premier to cut out what I didn't want.

**Reply by rkhamilton** [reply]
If you have x story prompts, and you want 20 seconds of each, then you need
20 seconds * framerate of your video (e.g. 24) = 480 video frames for each prompt.
You get one video frame every save_every iterations. So to get 480 video frames, you need 480*save_every iterations, for each prompt. So save_every is 10, that's 4800 iterations per prompt.

Build up your text prompts, and set iterations = number of prompts * 4800.

Hope that helps.
